service/storage_proxy.cc

```diff
@@ -5098,8 +5098,12 @@ class abstract_read_executor : public enable_shared_from_this<abstract_read_exec
                         && !data_resolver->any_partition_short_read()) {
                     tracing::trace(_trace_state, "Read stage is done for read-repair");
                     mlogger.trace("reconciled: {}", rr_opt->pretty_printer(_schema));
+                    auto validation_level = _proxy->local_db().get_config().enable_sstable_key_validation()
+                        ? mutation_fragment_stream_validation_level::clustering_key
+                        : mutation_fragment_stream_validation_level::token;
```

Comment:
I think this is confusing. The user might think that passing a high validation level will mean the stream is thoroughly validated, while in fact just a small subset will be checked somewhere deep down the stack, in a non-obvious place.

If we have a data source here which is prone to being corrupt, why not instantiate a full-blown validator here (conditionally).